src/packs/cli.rs

```diff
@@ -129,6 +133,10 @@ pub fn run() -> Result<(), Box<dyn std::error::Error>> {
             Ok(())
         }
         Command::Check { files } => checker::check_all(configuration, files),
+        Command::CheckContents { file } => {
+            configuration.file_source = FileSource::Stdin;
```

Comment:
There's one small issue with this approach I realized, which is that it doesn't work well with the experimental parser.

The experimental parser requires that *all* files in `configuration.included_files` are parsed in order to get the definition list.

This implementation suggests that for all files, we read them either from `Stdin` or from `Disk`.

One implementation sequencing that gets around this:
1. Change `get_all_references` to behave differently – to get *all* processed files with the experimental parser, and only the relevant ones with the regular parser. I think this is a bug currently with the experimental parser when passing in input files.
2. Remove `file_source` from `Configuration`, since some files will come from disk and some from stdin. 
3a. Change `check_all` to accept `files_from_disk` and `file_from_stdin` as separate parameters. `file_from_stdin` is only a single file rather than a vec of files because I assume there is no use case where multiple files are read in from stdin (let me know if that's not correct).
3b. Alternatively, `check_all` can accept a single list of files and a source. However, if we can't read multiple files from stdin, I like making that state unrepresentable.
4. `process_from_path` can accept a `FileSource` where to read a given file from.

What do you think?

Reply:
I fixed the issue related to (1) in this PR: https://github.com/alexevanczuk/packs/pull/87

Reply:
Sounds good, thanks @oleg-vinted :)